egs/wsj/s5/steps/online/nnet3/decode.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+# Copyright 2014  Johns Hopkins University (Author: Daniel Povey)
+# Apache 2.0
+
+# Begin configuration section.  
+stage=0
+nj=4
+cmd=run.pl
+max_active=7000
+threaded=false
+modify_ivector_config=false #  only relevant to threaded decoder.
+beam=15.0
+lattice_beam=6.0
+acwt=0.1   # note: only really affects adaptation and pruning (scoring is on
+           # lattices).
+post_decode_acwt=1.0  # can be used in 'chain' systems to scale acoustics by 10 so the
+                      # regular scoring script works.
+per_utt=false
+online=true  # only relevant to non-threaded decoder.
+do_endpointing=false
+do_speex_compressing=false
+scoring_opts=
+skip_scoring=false
+silence_weight=1.0  # set this to a value less than 1 (e.g. 0) to enable silence weighting.
+max_state_duration=40 # This only has an effect if you are doing silence
+  # weighting.  This default is probably reasonable.  transition-ids repeated
+  # more than this many times in an alignment are treated as silence.
+iter=final
+online_config=
+# End configuration section.
+
+echo "$0 $@"  # Print the command line for logging
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+if [ $# != 3 ]; then
+   echo "Usage: $0 [options] <graph-dir> <data-dir> <decode-dir>"
+   echo "... where <decode-dir> is assumed to be a sub-directory of the directory"
+   echo " where the models are, as prepared by steps/online/nnet2/prepare_online_decoding.sh"
```

Comment:
> In egs/wsj/s5/steps/online/nnet3/decode.sh
> <https://github.com/kaldi-asr/kaldi/pull/666#discussion_r58906478>:
>
> > +max_state_duration=40 # This only has an effect if you are doing silence
> > +  # weighting.  This default is probably reasonable.  transition-ids repeated
> > +  # more than this many times in an alignment are treated as silence.
> > +iter=final
> > +online_config=
> > +# End configuration section.
> > +
> > +echo "$0 $@"  # Print the command line for logging
> > +
> > +[ -f ./path.sh ] && . ./path.sh; # source the path.
> > +. parse_options.sh || exit 1;
> > +
> > +if [ $# != 3 ]; then
> > +   echo "Usage: $0 [options] <graph-dir> <data-dir> <decode-dir>"
> > +   echo "... where <decode-dir> is assumed to be a sub-directory of the directory"
> > +   echo " where the models are, as prepared by steps/online/nnet2/prepare_online_decoding.sh"
>
> I can copy/paste nnet2 script, maybe change it a bit to produce
> online.conf instead of conf/online_nnet2_decoding.conf. I can not test it
> on ivectors.
>
OK, we can test that.

> Still maybe nnet2/nnet3 script should be just one script.
>
I prefer the nnet3 stuff to be self-contained without unnecessary
dependency on the nnet2 directory.

> Also I always did not like how
> steps/online/nnet2/prepare_online_decoding.sh works because of it producing
> absolute paths. You can not move model and reuse in other place after that.
>
> There should be some sort of standardization on how models are bundled, so
> they can be reused from anywhere.
>
You're right, it's is not ideal, but let's mostly copy it for now-- this
probably isn't the time for a big rewrite.

Dan



> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/666/files/7bd2d2e8b3ef5934603c80d3c3d4dcddff5117bb#r58906478>
>